### PR TITLE
Feat/drop down menu

### LINKED ISSUE #3: Research and POC the best version for a drop down menu
Success criteria
- Multi - up to 3 level drop down menu, utilizing default bootstrap styling implemented
- Menu is hooked to Piranha CMS categories and content
- Menu is present as a .NET MVC re-usable and importable component.

### PR DESCRIPTION
**Pull request by mkyordanov**
Recursively generated drop-down navigation menu

### CODE REVIEW

File: Views/Shared/RecursiveNavigation.cshtml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+@model Piranha.Models.SitemapItem
+
+@if (@Model.Items.Count > 0)
+{
+    <li class="@(Model.ParentId == null ? "nav-item dropdown" : "")">
```
Comment (adautev) [inline Views/Shared/RecursiveNavigation.cshtml]: We don't want class="". Please fill in the whole class=nav-item or nothing.

File: Views/Shared/RecursiveNavigation.cshtml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <a class="@(Model.ParentId == null ? "nav-link dropdown-toggle" : "dropdown-item")" href="@WebApp.Url(Model)" data-toggle="@(Model.ParentId == null ? "dropdown" : "")">
```
Comment (adautev) [inline Views/Shared/RecursiveNavigation.cshtml]: Same as above.

File: Views/Shared/RecursiveNavigation.cshtml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            @Model.MenuTitle @(Model.ParentId == null ? "" : ">>")
+        </a>
+        <ul class=" @(Model.ParentId == null ? "dropdown-menu dropdown-submenu" : "submenu dropdown-menu")">
```
Comment (adautev) [inline Views/Shared/RecursiveNavigation.cshtml]: Same as above.

File: Views/Shared/RecursiveNavigation.cshtml (hunk 1 above)
Comment (mkyordanov) [reply Views/Shared/RecursiveNavigation.cshtml]: Done

File: Views/Shared/RecursiveNavigation.cshtml (hunk 2 above)
Comment (mkyordanov) [reply Views/Shared/RecursiveNavigation.cshtml]: Done

File: Views/Shared/RecursiveNavigation.cshtml (hunk 3 above)
Comment (mkyordanov) [reply Views/Shared/RecursiveNavigation.cshtml]: Done

**Pull request merged**